src/client/api/init.c

```diff
@@ -37,7 +37,7 @@ const struct daos_task_api dc_funcs[] = {
 	/** Management */
 	{dc_deprecated, 0},
 	{dc_deprecated, 0},
-	{dc_deprecated, 0},
+	{dc_mgmt_pool_list, sizeof(daos_mgmt_pool_list_t)},
```

Comment:
I'm not sure. I was following the pattern set by @kccain when he added the [original implementation of pool list](https://github.com/daos-stack/daos/commit/3b97fc1523d86fc9c8e0e088351cee41cb8872b0). I haven't really dug into it beyond that.

Given that I don't feel strongly either way, I'm happy to follow your guidance on the best way to implement this feature.

Reply:
@mchaarawi: Ping... Please advise on the recommended approach here. I'm getting ready to rebase all of this work on master, and would prefer to push that PR in as close to a final state as possible rather than iterating on it.

Also, I don't see much in the way of client API tests. Not particularly keen to invent all of that myself, but happy to work with someone who knows more about this side of things. I am planning to add ftests for this feature -- would that be acceptable in lieu of new client API unit tests?

Reply:
sorry i didn't know you were still looking for my review on this. as i mentioned, you should just avoid adding this change with the task api, otherwise you still need to update other structures to support that.
for client api tests, im not sure what you mean. i believe you should add some tests there:
https://github.com/daos-stack/daos/blob/master/src/tests/suite/daos_mgmt.c
but if you want to use ftest, that is fine too.

Reply:
> sorry i didn't know you were still looking for my review on this. as i mentioned, you should just avoid adding this change with the task api, otherwise you still need to update other structures to support that. 

OK, is the task API deprecated? I have a superficial understanding of the client API's workings, so as I said I was just following the existing patterns. If I didn't use the task API, would I basically just rework `daos_mgmt_list_pools()` to just call `dc_mgmt_pool_list()` directly? Is there any downside to doing it this way?

> for client api tests, im not sure what you mean. i believe you should add some tests there: https://github.com/daos-stack/daos/blob/master/src/tests/suite/daos_mgmt.c but if you want to use ftest, that is fine too.

Ah, OK. I was thinking of unit tests under `src/client/api/tests`. I'll just rework the suite test to use the re-added client API instead of the dmg helper.

Reply:
> OK, is the task API deprecated? I have a superficial understanding of the client API's workings, so as I said I was just following the existing patterns. 

no the task api is not deprecated. but to add the task API entry for this function, you still need to update:
https://github.com/daos-stack/daos/blob/mjmac/DAOS-15982/src/include/daos/task.h
to be safe.
otherwise you can just remove the task api entry for it, and do as you suggested:

>If I didn't use the task API, would I basically just rework `daos_mgmt_list_pools()` to just call `dc_mgmt_pool_list()` directly? Is there any downside to doing it this way?

either way is fine for me.